app/controllers/rosters_controller.rb

```diff
@@ -1,8 +1,9 @@
 class RostersController < ApplicationController
   # The default scaffold method, not the generic one
   # we wrote in ApplicationController.
-  before_action :find_roster, only: [:destroy, :edit, :update]
-  before_action :validate_admin_in_roster, only: %i(destroy edit update)
+  before_action :find_roster, only: %i(destroy edit update)
+  before_action :require_admin_in_roster, only: %i(destroy edit update)
+  before_action :require_admin, except: %i(destroy edit update)
```

Comment:
It occurs to me that it would be fine to write this as

```ruby
before_action :require_admin
before_action :require_admin_in_roster, only: %i(destroy edit update)
```

Would you like me to make that change?